### PR TITLE
Bugifx ql timeseries units

### LINKED ISSUE #104: Units on timeseries `.plot` and `.quantity` are inconsistent
When I extract a quantity from a STIX quicklook lightcurve using `.quantity`, the units are in `ct`. However, when I visualize that same data with `peek` or `plot`, the y-label says the units are `ct s-1 keV-1`. The latter appears to be hardcoded in the `plot` method.

### PR DESCRIPTION
**Pull request by samaloney**
* Fix units so those in plots match the units in data
* Fix bug data units did not match the actual data units
   * Data had ct s-1 keV-1 but was really ct keV-1 now data is really in ct s-1 keV-1 
* Reconstruct the QTables from the HDU header pair so get units from headers
* Update livetime correction to use same function as imaging with the new parameter values
* Update plot code to use more of sunpy TimeSeries functions
* Add columns keyword with useful defaults
* Update tests to check for correct quantities

Closes #104

```
from matplotlib import pyplot as plt
from sunpy.timeseries import TimeSeries
from stixpy.timeseries.quicklook import HKMaxi, QLBackground, QLLightCurve, QLVariance

ql_lc = TimeSeries(r"https://pub099.cs.technik.fhnw.ch/fits/L1/2020/05/06/QL/solo_L1_stix-ql-lightcurve_20200506_V02.fits")
ql_lc.quantity('4-10 keV').unit
# Unit("ct / (keV s)")

ql_lc.plot()
plt.show()
```
stix_ql_example

### CODE REVIEW

File: stixpy/timeseries/quicklook.py [hunk 1]
```diff
@@ -1,32 +1,49 @@
...
 
-def _lfrac(trigger_rate):
-    nin = trigger_rate / (1 - (trigger_rate * (tau+eta)))
-    return np.exp(-eta * nin) / (1 + tau * nin)
+    Returns
+    -------
+    QTable
+    """
+    header = hdupair.header
+    # TODD remove when python 3.9 and astropy 5.3.4 are dropped weird non-ascii error
```
Comment (DanRyanIrish) [inline stixpy/timeseries/quicklook.py:31]:
```suggestion
    # TODO remove when python 3.9 and astropy 5.3.4 are dropped weird non-ascii error
```

File: stixpy/timeseries/quicklook.py [hunk 2]
```diff
@@ -342,18 +295,16 @@ def _parse_hdus(cls, hdulist):
             The path to the file you want to parse.
         """
         header = hdulist[0].header
-        control = Table(hdulist[1].data)
-        header["control"] = control
-        data = Table(hdulist[2].data)
-        energies = Table(hdulist[4].data)
+        # control = _hdu_to_qtable(hdulist[1])
```
Comment (DanRyanIrish) [inline stixpy/timeseries/quicklook.py:298]: Should this be uncommented?  If not, why is it uncommented in `QLVariance`, below.

**Review by DanRyanIrish — COMMENTED** [review]
Where's the key change that actually scales the data by time correctly?

File: stixpy/timeseries/quicklook.py [hunk 3]
```diff
@@ -139,44 +123,38 @@ def _parse_file(cls, filepath):
...
+        data = _hdu_to_qtable(hdulist[2])
+        energies = _hdu_to_qtable(hdulist[4])
+        energy_delta = energies["e_high"] - energies["e_low"]
 
-        live_time = _lfrac(data["triggers"].reshape(-1) / (16 * data["timedel"] * u.s))
+        live_frac, *_ = get_livetime_fraction(data["triggers"].reshape(-1) / (16 * data["timedel"]))
 
-        data["counts"] = data["counts"] / (live_time.reshape(-1, 1) * energy_delta)
+        data["counts"] = data["counts"] / ((data['timedel'].to(u.s) * live_frac).reshape(-1, 1) * energy_delta)
```
Comment (samaloney) [inline stixpy/timeseries/quicklook.py:141]: @DanRyanIrish this is the fix before was not multiplying the integration time by the livefrac e.g. `live_time.reshape(-1, 1)` -> `(data['timedel'].to(u.s) * live_frac).reshape(-1, 1)`

File: stixpy/timeseries/quicklook.py [hunk 4]
```diff
@@ -342,18 +295,16 @@ def _parse_hdus(cls, hdulist):
...
+        # control = _hdu_to_qtable(hdulist[1])
+        data = _hdu_to_qtable(hdulist[2])
+        energies = _hdu_to_qtable(hdulist[4])
+        energy_delta = energies["e_high"] - energies["e_low"]
 
-        energy_delta = energies["e_high"] - energies["e_low"] << u.keV
+        live_frac, *_ = get_livetime_fraction(data["triggers"].reshape(-1) / data["timedel"])
 
-        live_time = _lfrac(data["triggers"].reshape(-1) / (16 * data["timedel"] * u.s))
+        data["counts"] = data["counts"] / ((data['timedel'].to(u.s) * live_frac).reshape(-1, 1) * energy_delta)
```
Comment (samaloney) [inline stixpy/timeseries/quicklook.py:305]: @DanRyanIrish this is the live time correction fix for BKG data

File: stixpy/timeseries/quicklook.py [hunk 5]
```diff
@@ -517,32 +455,31 @@ def _parse_hdus(cls, hdulist):
...
 
         try:
-            data["time"] = Time(header["date_obs"]) + TimeDelta(data["time"] * u.s)
+            data["time"] = Time(header["date_obs"]) + TimeDelta(data["time"])
         except KeyError:
-            data["time"] = Time(header["date-obs"]) + TimeDelta(data["time"] * u.s)
+            data["time"] = Time(header["date-obs"]) + TimeDelta(data["time"])
 
-        data["variance"] = data["variance"].reshape(-1) / (dE * data["timedel"])
+        data["variance"] = data["variance"].reshape(-1) * u.ct / (delta_energy * data["timedel"].to(u.s))
```
Comment (samaloney) [inline stixpy/timeseries/quicklook.py:474]: I'm actually not totally sure what units should be here TBH or if we should be correcting for live time.

**Comment by DanRyanIrish** [comment]
OK.  Well, as long as the tests pass and the plots look correct, I'm happy to approve this PR.

**Review by DanRyanIrish — APPROVED** [review]

File: stixpy/timeseries/quicklook.py [hunk 6]
```diff
@@ -206,8 +184,12 @@ def __repr__(self):
 
 
 class QLBackground(GenericTimeSeries):
-    """
-    Quicklook X-ray background detector time series.
+    r"""
+    Quicklook X-ray time series from the background detector.
+
+    Background monitoring is done in such way that counts from background detector are summed in five specified energy
```
Comment (hayesla) [inline stixpy/timeseries/quicklook.py:190]: again I'd say maybe just mention that this is a QL data product and the file

File: stixpy/timeseries/quicklook.py [hunk 7]
```diff
@@ -664,9 +601,9 @@ def _parse_hdus(cls, hdulist):
         data = Table(hdulist[2].data)
 
         try:
-            data["time"] = Time(header["date_obs"]) + TimeDelta(data["time"] * u.s)
+            data["time"] = Time(header["date_obs"]) + TimeDelta(data["time"] * u.cs)
         except KeyError:
-            data["time"] = Time(header["date-obs"]) + TimeDelta(data["time"] * u.s)
+            data["time"] = Time(header["date-obs"]) + TimeDelta(data["time"] * u.cs)
```
Comment (hayesla) [inline stixpy/timeseries/quicklook.py:606]: does `data["time"]` not already have a unit? i.e. should it not be 

```suggestion
            data["time"] = Time(header["date-obs"]) + TimeDelta(data["time"])
```

**Review by hayesla — APPROVED** [review]
once tests pass, all good :)

File: stixpy/timeseries/quicklook.py [hunk 8]
```diff
@@ -139,44 +123,38 @@ def _parse_file(cls, filepath):
...
         ----------
         hdulist :
         """
         header = hdulist[0].header
-        control = QTable(hdulist[1].data)
-        header["control"] = control
-        data = QTable(hdulist[2].data)
-        energies = QTable(hdulist[4].data)
-        energy_delta = energies["e_high"] - energies["e_low"] << u.keV
+        # control = _hdu_to_qtable(hdulist[1])
```
[the remark this answers is not in the document]
Comment (samaloney) [reply stixpy/timeseries/quicklook.py:134]: I want to keep these as will need in future PR

**Comment by samaloney** [comment]
Urgh - I don't understand the windows test fail on HK it happening at download time which is done in sunpy and the same thing works for the QL 😭

**Pull request merged**